### DISCUSSION TITLE
The prompt multiplies if If resize the emulator

### DISCUSSION THREAD

**Discussion by blueray453** [fish-shell/fish-shell#9831] [Q&A] [answered]
If I resize the terminal emulator then the prompt multiplies.

Is it only in my machine or is it a bug?

The following is the informative prompt:

![informative fish](https://github.com/fish-shell/fish-shell/assets/855297/357162bd-198b-46e3-81c9-f851510e67c2)

The following is the astronaut prompt:

![astronaut fish](https://github.com/fish-shell/fish-shell/assets/855297/c008e598-5673-40e5-87bf-73dcf1fa2bef)

**Comment by faho** [comment, marked as answer]
This is what happens when the terminal doesn't reflow the prompt on its own, and by the time fish has redrawn it instead it's already moved on.

That's an unfortunate limitation of the terminal model and there is nothing we can do about it.

**Reply by blueray453** [reply]
Do you know that the problem is solved for me.

I have a made a function like:

```
function _redraw_on_columns_change --on-variable COLUMNS
    clear
    commandline -f repaint
end

```
and put it in `conf.d`.
